### ISSUE TITLE
Extend AlwaysTrueOrFalseCondition to global scope

### ISSUE THREAD

**Issue by fmallet** [SonarSource/SonarJS#689]
The current implementation of [RSPEC-2583](https://jira.sonarsource.com/browse/RSPEC-2583) only consider function bodies. However, it should also be able to raise issues in the global scope.

To be able to fix this ticket we need configuration providing global variables available. Without it we produce FP.
Assuming this code executed with provided `Foo` variable, issue on first line is false positive
```
if (!Foo) { // always true condition
  var Foo = {};
}
```

**Comment by yassin-kammoun-sonarsource** [comment]
The rule was deprecated and eventually dropped #2339.
